src/gui/qgsmapcanvas.cpp

```diff
@@ -3270,12 +3268,12 @@ void QgsMapCanvas::readProject( const QDomDocument &doc )
   else
   {
     QgsDebugMsgLevel( QStringLiteral( "Couldn't read mapcanvas information from project" ), 2 );
-    if ( !project->viewSettings()->defaultViewExtent().isNull() )
+    if ( !QgsProject::instance()->viewSettings()->defaultViewExtent().isNull() )
```

Comment:
This should use mProject if set

Reply:
So also in line 3253 ?

Reply:
Yeah, everywhere the associated project is accessed

Reply:
ok, so in almost every location where ` QgsProject::instance()` is used in the `QgsMapCanvas` class.
In that case, in the scope of this PR, let me change only in this method.